hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/clean/CleanPlanActionExecutor.java

```diff
@@ -85,10 +86,18 @@ HoodieCleanerPlan requestClean(HoodieEngineContext context) {
 
       context.setJobStatus(this.getClass().getSimpleName(), "Generating list of file slices to be cleaned");
 
-      Map<String, List<HoodieCleanFileInfo>> cleanOps = context
-          .map(partitionsToClean, partitionPathToClean -> Pair.of(partitionPathToClean, planner.getDeletePaths(partitionPathToClean)), cleanerParallelism)
-          .stream()
-          .collect(Collectors.toMap(Pair::getKey, y -> CleanerUtils.convertToHoodieCleanFileInfoList(y.getValue())));
+      Map<String, List<HoodieCleanFileInfo>> cleanOps;
+      if (isDropPartition) {
+        cleanOps = context
+            .map(partitionsToClean, partitionPathToClean -> Pair.of(partitionPathToClean, planner.getDeletePartitionPaths(partitionPathToClean)), cleanerParallelism)
```

Comment:
from what I infer, we are just returning the partition path directory as part of List<HoodieCleanFileInfo> here and don't explicitly add every file group within the partition. since we store entries in metadata table keyed on partition path, we will just directly delete the entry and so add every file group info is not required? can you confirm my understanding.

Reply:
shouldn't we atleast fix CleanFileInfo to have another variable called isPartitionPath or something.